Json Schema Generator throws error in console

Steps:
Select an Infomodel > Generate code > Json Schema Generator
Actual Result:
_Error in console:_
[ERROR] java.lang.NullPointerException
            at org.eclipse.vorto.codegen.examples.jsonschema.tasks.template.ConfigurationValidationTemplate.getContent(ConfigurationValidationTemplate.java:60)
            at org.eclipse.vorto.codegen.examples.jsonschema.tasks.template.ConfigurationValidationTemplate.getContent(ConfigurationValidationTemplate.java:1)
            at org.eclipse.vorto.codegen.api.AbstractTemplateGeneratorTask.generate(AbstractTemplateGeneratorTask.java:24)
            at org.eclipse.vorto.codegen.api.ChainedCodeGeneratorTask.generate(ChainedCodeGeneratorTask.java:33)
            at org.eclipse.vorto.codegen.examples.jsonschema.JsonschemaGenerator.generateForConfiguration(JsonschemaGenerator.java:98)
            at org.eclipse.vorto.codegen.examples.jsonschema.JsonschemaGenerator.generateForFunctionblock(JsonschemaGenerator.java:81)
            at org.eclipse.vorto.codegen.examples.jsonschema.JsonschemaGenerator.generate(JsonschemaGenerator.java:51)
            at org.eclipse.vorto.codegen.ui.handler.CodeGeneratorInvocationHandler.evaluate(CodeGeneratorInvocationHandler.java:85)
            at org.eclipse.vorto.codegen.ui.handler.CodeGeneratorInvocationHandler.execute(CodeGeneratorInvocationHandler.java:60)
            at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:295)
            at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:90)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
            at java.lang.reflect.Method.invoke(Unknown Source)
            at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:56)
            at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:252)
            at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:234)
            at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
            at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
            at org.eclipse.core.commands.Command.executeWithChecks(Command.java:493)
            at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:486)
            at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
            at org.eclipse.ui.internal.handlers.LegacyHandlerService.executeCommand(LegacyHandlerService.java:343)
            at org.eclipse.ui.menus.CommandContributionItem.handleWidgetSelection(CommandContributionItem.java:830)
            at org.eclipse.ui.menus.CommandContributionItem.access$21(CommandContributionItem.java:816)
            at org.eclipse.ui.menus.CommandContributionItem$5.handleEvent(CommandContributionItem.java:806)
            at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
            at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4362)
            at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1113)
            at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4180)
            at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3769)
            at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
            at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
            at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
            at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
            at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:694)
            at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
            at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:606)
            at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
            at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:139)
            at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
            at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
            at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
            at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
            at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
            at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
            at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
            at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
            at java.lang.reflect.Method.invoke(Unknown Source)
            at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
            at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
            at org.eclipse.equinox.launcher.Main.run(Main.java:1515)

--- a/bundles/org.eclipse.vorto.codegen.ui/src/org/eclipse/vorto/codegen/ui/handler/CodeGeneratorInvocationHandler.java
+++ b/bundles/org.eclipse.vorto.codegen.ui/src/org/eclipse/vorto/codegen/ui/handler/CodeGeneratorInvocationHandler.java
@@ -113,7 +113,9 @@ public class CodeGeneratorInvocationHandler extends AbstractHandler {
 			InformationModel infoModel = (InformationModel) model;
 			for (FunctionblockProperty property : infoModel.getProperties()) {
 				// for syntax and parsing errors
-				if (property.getType().eResource().getErrors().size() > 0) {
+				if (property.getType() != null && property.getType().eResource() != null && 
+						property.getType().eResource().getErrors() != null && 
+						property.getType().eResource().getErrors().size() > 0) {
 					return false;
 				}
 

--- a/example-generators/org.eclipse.vorto.codegen.examples.jsonschema/src/org/eclipse/vorto/codegen/examples/jsonschema/JsonschemaGenerator.java
+++ b/example-generators/org.eclipse.vorto.codegen.examples.jsonschema/src/org/eclipse/vorto/codegen/examples/jsonschema/JsonschemaGenerator.java
@@ -65,29 +65,43 @@ public class JsonschemaGenerator implements IVortoCodeGenerator {
 			String targetPath, 
 			String jsonFileExtension,
 			IGeneratedWriter outputter) {
-		
-		for (Operation op : fb.getOperations()) {
-			generateForOperation(
-					op, 
-					targetPath,
-					jsonFileExtension, 
-					outputter);
+		if (fb == null) {
+			throw new RuntimeException("fb is null");
 		}
-
-		Configuration configuration = fb.getConfiguration();
-		Status status = fb.getStatus();
-		Fault fault = fb.getFault();
 		
-		generateForConfiguration(configuration, targetPath, jsonFileExtension, outputter);
-		generateForStatus(status, targetPath, jsonFileExtension, outputter);
-		generateForFault(fault, targetPath, jsonFileExtension, outputter);
-
-		for (Event event : fb.getEvents()) {
-			generateForEvent(
-					event, 
-					targetPath, 
-					jsonFileExtension, 
-					outputter);
+		if (fb.getOperations() != null) {
+			for (Operation op : fb.getOperations()) {
+				generateForOperation(
+						op, 
+						targetPath,
+						jsonFileExtension, 
+						outputter);
+			}
+		}
+		
+		Configuration configuration = fb.getConfiguration();
+		if (configuration != null) {
+			generateForConfiguration(configuration, targetPath, jsonFileExtension, outputter);
+		}
+		
+		Status status = fb.getStatus();
+		if (status != null) {
+			generateForStatus(status, targetPath, jsonFileExtension, outputter);
+		}
+		
+		Fault fault = fb.getFault();
+		if (fault != null) {
+			generateForFault(fault, targetPath, jsonFileExtension, outputter);
+		}
+		
+		if (fb.getEvents() != null) {
+			for (Event event : fb.getEvents()) {
+				generateForEvent(
+						event, 
+						targetPath, 
+						jsonFileExtension, 
+						outputter);
+			}
 		}
 	}
 

